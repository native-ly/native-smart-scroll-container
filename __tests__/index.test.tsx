import React from 'react'
import { View } from 'react-native'
import { render, fireEvent } from '@testing-library/react-native'

import SmartScrollContainer from '../src'

describe('SmartScrollContainer', () => {
  it('should set scroll state automatically to enabled', () => {
    const { getByTestId } = render(
      <SmartScrollContainer testID="scroll-container" style={{ height: 500 }}>
        <View style={{ height: 1600 }} />
      </SmartScrollContainer>
    )

    expect(getByTestId('scroll-container').props.scrollEnabled).toBe(true)
  })

  it('should set scroll state automatically to disabled', () => {
    const { getByTestId } = render(
      <SmartScrollContainer testID="scroll-container" style={{ height: 500 }}>
        <View style={{ height: 100 }} />
      </SmartScrollContainer>
    )

    expect(getByTestId('scroll-container').props.scrollEnabled).toBe(false)
  })

  it('should render container with force disabled scroll', () => {
    const { getByTestId } = render(
      <SmartScrollContainer
        testID="scroll-container"
        scrollEnabled={true}
        style={{ height: 500 }}
      >
        <View style={{ height: 1600 }} />
      </SmartScrollContainer>
    )

    expect(getByTestId('scroll-container').props.scrollEnabled).toBe(true)
  })

  it('should render container with force enabled scroll', () => {
    const { getByTestId } = render(
      <SmartScrollContainer
        testID="scroll-container"
        scrollEnabled={true}
        style={{ height: 500 }}
      >
        <View style={{ height: 1600 }} />
      </SmartScrollContainer>
    )

    expect(getByTestId('scroll-container').props.scrollEnabled).toBe(true)
  })

  it('should render horizontal container', () => {
    const { getByTestId } = render(
      <SmartScrollContainer
        testID="scroll-container"
        horizontal
        style={{ width: 500 }}
      >
        <View style={{ width: 1600 }} />
      </SmartScrollContainer>
    )

    expect(getByTestId('scroll-container').props.scrollEnabled).toBe(true)
  })
})
