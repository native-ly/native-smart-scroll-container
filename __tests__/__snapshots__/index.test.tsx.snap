// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartScrollContainer should adjust to layout change 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onScroll={[MockFunction]}
  scrollEnabled={false}
  style={
    Object {
      "height": 860,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 1661,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`SmartScrollContainer should render container with enabled horizontal scroll 1`] = `
<RCTScrollView
  horizontal={true}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  scrollEnabled={true}
  style={
    Object {
      "height": 543,
    }
  }
  testID="scroll-container"
>
  <View>
    <View
      style={
        Object {
          "height": 1234,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`SmartScrollContainer should render container with force disabled scroll 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  scrollEnabled={false}
  style={
    Object {
      "height": 480,
    }
  }
  testID="scroll-container"
>
  <View>
    <View
      style={
        Object {
          "height": 1597,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`SmartScrollContainer should render container with force enabled scroll 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  scrollEnabled={true}
  style={
    Object {
      "height": 760,
    }
  }
  testID="scroll-container"
>
  <View>
    <View
      style={
        Object {
          "height": 1269,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`SmartScrollContainer should set scroll state automatically to disabled 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  scrollEnabled={false}
  style={
    Object {
      "height": 680,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 360,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`SmartScrollContainer should set scroll state automatically to enabled 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  scrollEnabled={false}
  style={
    Object {
      "height": 500,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 1600,
        }
      }
    />
  </View>
</RCTScrollView>
`;
